modules/flyte/storage.tf

```diff
@@ -1,11 +1,30 @@
+resource "azurerm_storage_account" "flyte" {
+  name                     = join("", [replace(var.deploy_id, "/[_-]/", ""), "flyte"])
+  location                 = var.resource_group_location
+  resource_group_name      = var.resource_group_name
+  account_kind             = "StorageV2"
+  account_tier             = var.storage_account_tier
+  account_replication_type = var.storage_account_replication_type
+  access_tier              = "Hot"
+  min_tls_version          = "TLS1_2"
+  tags                     = var.tags
+  is_hns_enabled           = true
+
+  lifecycle {
+    ignore_changes = [
+      tags
+    ]
+  }
+}
+
 resource "azurerm_storage_container" "flyte_metadata" {
   name                  = "${var.deploy_id}-flyte-metadata"
-  storage_account_name  = var.azurerm_storage_account_name
+  storage_account_name  = azurerm_storage_account.flyte.name
```

Comment:
same as before, you can iterate through a single `resource "azurerm_storage_container"`